src/commands/external.rs

```diff
@@ -0,0 +1,56 @@
+use crate::commands::plugins::get_spin_plugins_directory;
+
+use anyhow::{anyhow, Result};
+use spin_plugins::version_check::check_plugin_spin_compatibility;
+use std::{collections::HashMap, env, path::Path};
+use tokio::process::Command;
+use tracing::log;
+
+/// Executes a Spin plugin as a subprocess, expecting the first argument to
+/// indicate the plugin to execute. Passes all subsequent arguments on to the
+/// subprocess.
+pub async fn execute_external_subcommand(args: Vec<String>) -> Result<()> {
+    let plugin_name = args.first().ok_or_else(|| anyhow!("Expected subcommand"))?;
+    let plugins_dir = get_spin_plugins_directory()?;
+    check_plugin_spin_compatibility(plugin_name, env!("VERGEN_BUILD_SEMVER"), &plugins_dir)?;
```

Comment:
It's a bit non-obvious to me where the existence check happens - it clearly does because we get a nice message, but it's not obvious in the code flow.

Reply:
broke it out into steps to make it clearer